play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -136,6 +136,130 @@ export class ChatRoom extends PSRoom {
+					let formatId = toID(row.formatid);
+					if (!formatTargeting ||
+						formats[formatId] ||
+						gens[formatId.slice(0, 4)] ||
+						(gens['gen6'] && formatId.substr(0, 3) !== 'gen')) {
+						buffer += '<tr>';
+					} else {
+						buffer += '<tr class="hidden">';
+						hiddenFormats.push(window.BattleLog.escapeFormat(formatId));
+					}
+
+					// Validate all the numerical data
+					let values = [row.elo, row.rpr, row.rprd, row.gxe, row.w, row.l, row.t];
+					for (let value of values) {
+						if (typeof value !== 'number' && typeof value !== 'string' ||
+							isNaN(value as number)) return this.receiveLine(['raw', 'Error: corrupted ranking data']);
+					}
+
+					buffer += `<td> ${BattleLog.escapeFormat(formatId)} </td><td><strong> ${Math.round(row.elo)} </strong></td>'`;
+					if (row.rprd > 100) {
+						// High rating deviation. Provisional rating.
+						buffer += `<td>&ndash;</td>`;
+						buffer += `<td><span><em> ${Math.round(row.rpr)} <small> &#177; ${Math.round(row.rprd)} </small></em> <small>(provisional)</small></span></td>`;
+					} else {
+						let gxe = Math.round(row.gxe * 10);
+						buffer += `<td> ${Math.floor(gxe / 10)} <small> ${(gxe % 10)}%</small></td>`;
+						buffer += `<td><em> ${Math.round(row.rpr)} <small> &#177; ${Math.round(row.rprd)}</small></em></td>`;
+					}
+					let N = parseInt(row.w, 10) + parseInt(row.l, 10) + parseInt(row.t, 10);
+					let COIL_B = undefined;
+
+					// Uncomment this after LadderRoom logic is implemented
+					// COIL_B = LadderRoom?.COIL_B[formatId];
+
+					if (COIL_B) {
+						buffer += `<td> ${Math.round(40.0 * parseFloat(row.gxe) * 2.0 ** (-COIL_B / N))} </td>`;
+					} else {
+						buffer += '<td>--</td>';
+					}
+					buffer += `<td> ${row.w} </td><td> ${row.l} </td><td> ${N} </td>`;
+					if (suspect) {
+						if (typeof row.suspect === 'undefined') {
+							buffer += '<td>--</td>';
+						} else {
+							buffer += '<td>';
+							buffer += (row.suspect ? "Yes" : "No");
+							buffer += '</td>';
+						}
+					}
+					buffer += '</tr>';
+				}
+				if (hiddenFormats.length) {
+					if (hiddenFormats.length === data.length) {
+						buffer += '<tr class="no-matches"><td colspan="8"><em>This user has not played any ladder games that match "' + BattleLog.escapeHTML(Object.keys(gens).concat(Object.keys(formats)).join(', ')) + '".</em></td></tr>';
+					}
+					buffer += `<tr><td colspan="8"><button name="showOtherFormats"> ${hiddenFormats.slice(0, 3).join(', ') + (hiddenFormats.length > 3 ? ' and ' + String(hiddenFormats.length - 3) + ' other formats' : '')} not shown</button></td></tr>'`;
+				}
+				let userid = toID(targets[0]);
+				let registered = PS.user.registered;
+				if (registered && PS.user.userid === userid) {
+					buffer += '<tr><td colspan="8" style="text-align:right"><a href="//' + PS.routes.users + '/' + userid + '">Reset W/L</a></tr></td>';
+				}
+				buffer += '</table></div>';
+				this.receiveLine(['raw', buffer]);
+			}, 'text');
```

Comment:
```suggestion
			});
```

to go with the `PSLoginServer` change above.